Sources/Prometheus/PrometheusCollectorRegistry.swift

```diff
@@ -82,9 +82,25 @@ public final class PrometheusCollectorRegistry: Sendable {
     }
 
     private let box = NIOLockedValueBox([String: Metric]())
+    private let dedupTypeHelpPerMetricNameOnEmit: Bool
 
-    /// Create a new collector registry
-    public init() {}
+    /// Creates a new PrometheusCollectorRegistry with default configuration.
+    ///
+    /// Uses deduplication for TYPE and HELP lines according to Prometheus specifications,
+    /// where only one TYPE and HELP line is emitted per metric name regardless of label sets.
+    public init() {
+        self.dedupTypeHelpPerMetricNameOnEmit = true
```

Comment:
Additional note: Since the last release, we have not allowed reusing the same metric name for distinct label sets, hence this new default should not break anything.